packages/bundler-plugin-core/src/utils/Output.ts

```diff
@@ -33,14 +39,21 @@ class Output {
   constructor(userOptions: NormalizedOptions) {
     this.version = "1";
     this.userOptions = userOptions;
+    this.internalOptions = {
+      frozenBundleName: false,
+      frozenPluginDetails: false,
```

Comment:
I don't like that we expose these internals to be set.

For `frozenPluginDetails` `frozenBundleName`, we can just add a setter for `this.plugin` and `this.bundleName` that takes an arg to freeze the variable after being set. This means that any followup calls to setter will not (we can even have a return value that indicates if something was overwritten or not).

So usage:

```ts
output.setBundleName(bundleName, { frozen: true });

// won't work - bundleName is still the one above
output.setBundleName(anotherName);

// feel free to use { name: pluginName, version: pluginVersion }
output.setPlugin(pluginName, pluginVersion, { frozen: true });
```

Also then does it make sense to freeze by default first time it's set?